### PR TITLE
Extensions Section

### PR DESCRIPTION
**Pull request by phazonoverload**
Review question:

There are subtle differences to how each sandboxed API extension type are registered compared to their non-sandboxed counterparts. I've included this at the bottom of each type page. Is this the right place for it? Or should it be in the sandbox page? I'd prefer to not duplicate unless we can make it a reusable partial that makes sense in both contexts.

### CODE REVIEW

File: content/10.extensions/0.overview.md [hunk 1]
```diff
@@ -1,6 +1,67 @@
...
 title: Overview
-description:
+description: Extensions are used to extend the functionality of Directus.
 ---
 
 # Extensions Overview
+
+Directus has been built to be extensible - both allowing for powerful enhancements to be built for single projects, and for publishing in the [Directus Marketplace](/extensions/marketplace).
+
+Extensions in Directus run within the same environment as the main application, allowing them to leverage existing access to underlying [services](/extensions/api-extensions/services) and [UI components](/extensions/app-extensions/ui-extensions).
```
Comment (hola-soy-milk) [inline content/10.extensions/0.overview.md]: The UI components link here is taking me to an empty page. Maybe it wasn't intended to be shown?

File: content/10.extensions/1.quickstart.md [hunk 2]
```diff
@@ -1,6 +1,54 @@
...
+Follow the steps in the [Create a Project](/getting-started/create-a-project) guide to set up your project locally. This `docker-compose.yml` file will set up a local volume for extensions to be loaded from. This directory exists on your local filesystem and is also mounted into the Docker container.
+
+Add the following to the `environment` section of your `docker-compose.yml` file to automatically reload your extensions when they are rebuilt:
+
+```yaml
+EXTENSIONS_AUTO_RELOAD: true
+```
+
+::callout{type="info" title="Restarting Directus"}
+When changing the `docker-compose.yml` file, you will need to restart Directus by pressing `Ctrl/Cmd+C` and then run `docker compose up` again.
```
Comment (hola-soy-milk) [inline content/10.extensions/1.quickstart.md]: Do we want to specify that this can also be done from the Docker app itself?

File: content/10.extensions/0.overview.md [hunk 3]
```diff
@@ -1,6 +1,67 @@
...
+
+[Themes](/extensions/app-extensions/themes) are used to style the Data Studio. They can be used to change the colors, fonts, and other visual elements of the Data Studio.
+
+## API Extensions
+
+[API Extensions](/extensions/api-extensions) extend the functionality of the API.
+
+### Hooks
+
+[Hooks](/extensions/api-extensions/hooks) allow running code when events occur within Directus. Events are triggered on schedules, database events, schedules, or during the Directus application lifecycle.
```
Comment (hola-soy-milk) [inline content/10.extensions/0.overview.md]:
```suggestion
[Hooks](/extensions/api-extensions/hooks) allow running code when events occur within Directus. Events are triggered on schedules, database events, or during the Directus application lifecycle.
```

File: content/10.extensions/2.api-extensions/1.hooks.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+| `schema`         | The current API schema in use.      |
+| `accountability` | Information about the current user. |
+
+### Filter Events
+
+| Name                           | Payload                               | Meta                                        |
+| ------------------------------ | ------------------------------------- | ------------------------------------------- |
+| `websocket.message`            | The message sent over the WebSocket.  |                                             |
+| `request.not_found`            | `false`                               | `request`, `response`                       |
+| `request.error`                | The request errors.                   | --                                          |
```
Comment (hola-soy-milk) [inline content/10.extensions/2.api-extensions/1.hooks.md]: Is the "--" here a string?

File: content/_partials/extension-hook-footguns.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Directus reads system collections to operate. Be careful when modifying the output of system collection `read` or `query` events. Also ensure not the diretly or indirectly emit the same event your hook is handling or you will create an infinite loop.
```
Comment (hola-soy-milk) [inline content/_partials/extension-hook-footguns.md]:
```suggestion
Directus reads system collections to operate. Be careful when modifying the output of system collection `read` or `query` events. Also ensure not to directly or indirectly emit the same event your hook is handling or you will create an infinite loop.
```

File: content/_partials/extension-hook-exceptions.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+::callout{type="info" title="System Collection Exceptions"}
+
+The `collections` and `fields` system collections does not emit a `read` event. The `files` collection does not emit a `create` or `update` event on file upload. The `relations` collection does not emit a `delete` event.
```
Comment (hola-soy-milk) [inline content/_partials/extension-hook-exceptions.md]:
```suggestion
The `collections` and `fields` system collections do not emit a `read` event. The `files` collection does not emit a `create` or `update` event on file upload. The `relations` collection does not emit a `delete` event.
```

File: content/10.extensions/2.api-extensions/1.hooks.md [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+
+```js
+export default ({ schedule }) => {
+	schedule('*/15 * * * *', () => {
+		console.log('15 minutes have passed.');
+	});
+};
+```
+
+The `schedule` event takes a cron string as the first argument and a callback function as the second argument. The cron string follows the following format:
```
Comment (hola-soy-milk) [inline content/10.extensions/2.api-extensions/1.hooks.md:228]: Do we want to link to cron docs?

File: content/10.extensions/2.api-extensions/2.endpoints.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+| ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `services`  | All API internal services.                                                                                                                              |
+| `database`  | Knex instance that is connected to the current database.                                                                                                |
+| `getSchema` | Async function that reads the full available schema for use in services.                                                                                |
+| `env`       | Parsed environment variables.                                                                                                                           |
+| `logger`    | [Pino](https://github.com/pinojs/pino) instance.                                                                                                        |
+| `emitter`   | [Event emitter](https://github.com/directus/directus/blob/main/api/src/emitter.ts) instance that can be used to emit custom events for other extensions |
```
Comment (hola-soy-milk) [inline content/10.extensions/2.api-extensions/2.endpoints.md]:
```suggestion
| `emitter`   | [Event emitter](https://github.com/directus/directus/blob/main/api/src/emitter.ts) instance that can be used to emit custom events for other extensions. |
```

File: content/10.extensions/2.api-extensions/3.operations.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+```js
+export default {
+	id: 'custom',
+	name: 'Custom',
+	icon: 'box',
+	description: 'This is my custom operation!',
+	overview: ({ text }) => [
+		{
+			label: 'Text',
+			text: text,
```
Comment (hola-soy-milk) [inline content/10.extensions/2.api-extensions/3.operations.md:30]:
```suggestion
			text,
```

Do we want to use property value shorthand here?

File: content/10.extensions/2.api-extensions/4.services.md [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+```js
+const data = await fieldsService.updateField('collection_name', 'field_name', {
+	meta: {
+		icon: 'title',
+	},
+});
+```
+
+::callout{type="info"}
+You cannot update the field name via Directus after creation.
```
Comment (hola-soy-milk) [inline content/10.extensions/2.api-extensions/4.services.md]: Would it make sense to elaborate on when a field name can be edited? Or probably not, if I remember correctly.

Comment (hola-soy-milk) [reply content/10.extensions/2.api-extensions/4.services.md]: I think the wording here is throwing me off a little. What do you think about something like this?

```suggestion
You cannot update the field name via Directus after creating the field.
```

File: content/10.extensions/3.app-extensions/1.interfaces.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+:partial{content="extensions-app"}
+
+## Interface Entrypoint
+
+The `index.js` or `index.ts` file exports an object that is read by Directus. It contains properties that control how an interface is displayed within menus, it’s what types it supports, what configurable options will be available to users, and the actual Vue component that will be loaded.
+
+### Entrypoint Example
+
+```vue
```
Comment (hola-soy-milk) [inline content/10.extensions/3.app-extensions/1.interfaces.md:21]: Oddly enough, the syntax highlighting isn't working here. Wonder if JS would do the trick since there's no markup?

File: content/10.extensions/3.app-extensions/1.interfaces.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+| `group`               | string    | The group this interface is shown at when creating a field. Accepts `standard`, `selection`, `relational`, `presentation`, `group`, or `other`. Defaults to `other`.               |
+| `relational`          | boolean   | Indicates if this a relational interface.                                                                                                                                          |
+| `recommendedDisplays` | array     | A list of display names which are recommended to be used with this interface.                                                                                                      |
+| `options`             | array     | component                                                                                                                                                                          | When an array, options contains user-configurable fields that are set when creating or editing the interface.
```
Comment (hola-soy-milk) [inline content/10.extensions/3.app-extensions/1.interfaces.md]: Does "component" need elaborating here? Or a full stop?

File: content/10.extensions/3.app-extensions/3.layouts.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+### Props
+
+The layout component will be passed all user configuration options from the entrypoint file. It will also receive the following props: 
+
+| Prop            | Type     | Description                                                  |
+| --------------- | -------- | ------------------------------------------------------------ |
+| `collection`    | string   | The current collection's name.                               |
+| `selection`     | array    | Any currently selected items.                                |
+| `layoutOptions` | object   | The user's currently saved layout options.                   |
+| `layoutQuery`   | object   | The user's layout query parameters. (e.g., sort, limit, etc) |
```
Comment (hola-soy-milk) [inline content/10.extensions/3.app-extensions/3.layouts.md]:
```suggestion
| `layoutQuery`   | object   | The user's layout query parameters. (e.g., sort, limit, etc). |
```

File: content/10.extensions/3.app-extensions/4.panels.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+| `name`        | string    | The displayed name for this panel in the Data Studio.                                                                     |
+| `icon`        | string    | An icon name from the [Google Material Icons set](https://fonts.google.com/icons). Supports filled and outlined variants. |
+| `description` | string    | A description of this panel shown in the Data Studio. Maximum 80 characters.                                              |
+| `component`   | component | A reference to the Vue component rendered in the dashboard.                                                               |
+| `minWidth`    | number    | Smallest number of grid units in the dashboard.                                                                           |
+| `minHeight`   | number    | Smallest number of grid units in the dashboard.                                                                           |
+| `options`     | array     | component                                                                                                                 | When an array, options contains user-configurable fields that are set when creating or editing the panel.
```
Comment (hola-soy-milk) [inline content/10.extensions/3.app-extensions/4.panels.md]: Should "component" here be capitalised or expanded upon?

File: content/10.extensions/3.app-extensions/7.ui-library.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
+---
+description: Directus offers globally-registered UI components for building extensions.
+---
+
+# UI Library
+
+Directus offers reusable UI components for building extensions, providing a consistent user experience across the Directus Data Studio. These components are used by Directus itself and can be utilized in any extension without explicit imports.
+
+Explore the available components in our [Directus Components Playground](https://components.directus.io/). Here, you can
```
Comment (hola-soy-milk) [inline content/10.extensions/3.app-extensions/7.ui-library.md:9]: Should this be an external link?

File: content/10.extensions/5.marketplace/1.publishing.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
+---
+description:
+---
+
+# Publishing Extensions
+
+Directus hosts a registry mirroring npm, with enhanced search/filtering and security features like blocking/delisting. The Directus Extensions Registry is updated every few hours, and only the latest version of an extension is available in the Marketplace.
```
Comment (hola-soy-milk) [inline content/10.extensions/5.marketplace/1.publishing.md:7]: Probably overkill to link NPM right

File: content/10.extensions/5.marketplace/1.publishing.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+The extension title displayed in the extension detail page is based on your npm package name and the following
+processing:
+
+- The `directus-extension-` prefix is removed, if present.
+- The package's scope is removed, if present.
+- The title is parsed by the [Title Formatter](https://github.com/directus/format-title) used throughout the Data Studio.
+
+### README
+
+Your npm package's `README.md` file is shown on the extension detail page. This can be updated by releasing a new version of your package to npm.
```
Comment (hola-soy-milk) [inline content/10.extensions/5.marketplace/1.publishing.md:48]: I'd love to have a link to an example if we don't already have a link to one I missed

**Review by hola-soy-milk — CHANGES_REQUESTED** [review]
This is excellent work. 

I'm trying to really stick to the goal of the docs being a reference, so I think the structure you propose is best. It might however be good to make that note that there are subtle differences in the sandbox page.

Side note: Having API and App next to each other as categories of extensions is a bit hard to parse sometimes. I wonder if there's a way we can make this easier? 
I'm of course not suggesting to rename one of them, just thinking out loud.

File: content/10.extensions/0.overview.md (hunk 3 above)
Comment (hola-soy-milk) [reply content/10.extensions/0.overview.md]: Repetition of the word schedules

File: content/10.extensions/0.overview.md (hunk 1 above)
Comment (phazonoverload) [reply content/10.extensions/0.overview.md]: It's a broken link. Will fix it later. Should be UI Library

File: content/10.extensions/1.quickstart.md (hunk 2 above)
Comment (phazonoverload) [reply content/10.extensions/1.quickstart.md]: Or if we want to be even more generic, we can just say they have to restart the docker container.

File: content/10.extensions/2.api-extensions/1.hooks.md (hunk 4 above)
Comment (phazonoverload) [reply content/10.extensions/2.api-extensions/1.hooks.md]: It needs removing. It means none.

File: content/10.extensions/1.quickstart.md (hunk 2 above)
Comment (hola-soy-milk) [reply content/10.extensions/1.quickstart.md]: Yes I think I like that best

File: content/10.extensions/2.api-extensions/1.hooks.md (hunk 7 above)
Comment (phazonoverload) [reply content/10.extensions/2.api-extensions/1.hooks.md:228]: I don't think so. Showing the structure is gonna be enough for most.

File: content/10.extensions/2.api-extensions/3.operations.md (hunk 9 above)
Comment (phazonoverload) [reply content/10.extensions/2.api-extensions/3.operations.md:30]: No we want to be explicit. Your suggestion is correct.

File: content/10.extensions/2.api-extensions/1.hooks.md (hunk 7 above)
Comment (hola-soy-milk) [reply content/10.extensions/2.api-extensions/1.hooks.md:228]: Makes sense!

File: content/10.extensions/2.api-extensions/4.services.md (hunk 10 above)
Comment (phazonoverload) [reply content/10.extensions/2.api-extensions/4.services.md]: THE field but I'm happy with this suggestion

File: content/10.extensions/3.app-extensions/1.interfaces.md (hunk 12 above)
Comment (phazonoverload) [reply content/10.extensions/3.app-extensions/1.interfaces.md]: It's an array or a component. I've got in a post-beta issue to explain how to do options components.

File: content/10.extensions/3.app-extensions/4.panels.md (hunk 14 above)
Comment (phazonoverload) [reply content/10.extensions/3.app-extensions/4.panels.md]: Oh actually the formatting is just messed up. Should be array | component

File: content/10.extensions/3.app-extensions/7.ui-library.md (hunk 15 above)
Comment (phazonoverload) [reply content/10.extensions/3.app-extensions/7.ui-library.md:9]: Whatcha mean?

File: content/10.extensions/5.marketplace/1.publishing.md (hunk 16 above)
Comment (phazonoverload) [reply content/10.extensions/5.marketplace/1.publishing.md:7]: Yeah

File: content/10.extensions/5.marketplace/1.publishing.md (hunk 17 above)
Comment (phazonoverload) [reply content/10.extensions/5.marketplace/1.publishing.md:48]: We don't but we can make one.

File: content/10.extensions/3.app-extensions/4.panels.md (hunk 14 above)
Comment (hola-soy-milk) [reply content/10.extensions/3.app-extensions/4.panels.md]: Oh good to know, thanks!

File: content/10.extensions/3.app-extensions/7.ui-library.md (hunk 15 above)
Comment (hola-soy-milk) [reply content/10.extensions/3.app-extensions/7.ui-library.md:9]: I've noticed external links look like in the screenshot. 
image

Seeing as this is outside docs, was wondering if we should have it look like this. Dunno how to tbh. Maybe it's a domain thing.

File: content/10.extensions/3.app-extensions/1.interfaces.md (hunk 11 above)
Comment (phazonoverload) [reply content/10.extensions/3.app-extensions/1.interfaces.md:21]: Or we just need to load the language into the syntax highlighter

File: content/10.extensions/3.app-extensions/7.ui-library.md (hunk 15 above)
Comment (phazonoverload) [reply content/10.extensions/3.app-extensions/7.ui-library.md:9]: Oh I'll check the pattern as I built it heh. Might be "directus" in url.

File: content/10.extensions/3.app-extensions/1.interfaces.md (hunk 11 above)
Comment (hola-soy-milk) [reply content/10.extensions/3.app-extensions/1.interfaces.md:21]: It's possible! But I could swear the code sample lower down was working..

Comment (hola-soy-milk) [reply content/10.extensions/3.app-extensions/1.interfaces.md:21]: [Yes, can confirm this one lower down is working ](https://github.com/directus/docs/pull/27/files/d17515b5aa3aafc028918d8fcc9c2bc29cec3866#diff-7d737a488dca6dd213a0102e3879988e3225eab210bd09ba75886c8debba8a41R74)

image

File: content/10.extensions/3.app-extensions/7.ui-library.md (hunk 15 above)
Comment (phazonoverload) [reply content/10.extensions/3.app-extensions/7.ui-library.md:9]: This is fine, but fyi the rule is for links not to contain one of the following:

- directus.io
- directus.cloud
- localhost
- 127.0.0.1

In main.scss

File: content/10.extensions/5.marketplace/1.publishing.md (hunk 17 above)
Comment (phazonoverload) [reply content/10.extensions/5.marketplace/1.publishing.md:48]: Adding to backlog

**Comment by phazonoverload** [comment]
@connorwinston do you mind taking a look at this? The UiAsideNav in ExtensionListing.global.vue is causing issues. I've confirmed that data._path exists for these pages.
